crates/symbolicator/src/cache.rs

```diff
@@ -421,8 +442,38 @@ impl Cache {
     pub fn tempfile(&self) -> io::Result<NamedTempFile> {
         match self.tmp_dir {
             Some(ref path) => {
-                std::fs::create_dir_all(path)?;
-                Ok(tempfile::Builder::new().prefix("tmp").tempfile_in(path)?)
+                // The `cleanup` process could potentially remove the parent directories we are
+                // operating in, so be defensive here and retry the fs operations.
+                const MAX_RETRIES: usize = 2;
+                let mut retries = 0;
+                loop {
+                    retries += 1;
+
+                    if let Err(e) = std::fs::create_dir_all(path) {
+                        sentry::with_scope(
+                            |scope| scope.set_extra("path", path.display().to_string().into()),
+                            || tracing::error!("Failed to create cache directory: {:?}", e),
+                        );
+                        if retries >= MAX_RETRIES {
```

Comment:
This will make this whole function to be called only twice, not thrice. Which is technically incorrect, as it should "retry" N times (2), so it should be called 3 times.

`if retries > MAX_RETRIES` will make it be called 3 times.

Reply:
agreed. naming this is hard :-D